packages/augur-core/source/contracts/reporting/Affiliates.sol

```diff
@@ -10,9 +11,15 @@ contract Affiliates {
     // Maps an account to the referral account. Used to apply affiliate fees on settlement.
     mapping (address => address) public referrals;
 
-    uint256 constant internal AFFILIATE_VALIDATOR_GAS = 30000;
+    // Mapping of valid Affiliate Validators
+    mapping (address => bool) public affiliateValidators;
 
-    bytes4 constant internal AFFILIATE_VALIDATOR_SELECTOR = bytes4(keccak256("validateReference(address,address)"));
+    function createAffiliateValidator() public returns (AffiliateValidator) {
+        AffiliateValidator _affiliateValidator = new AffiliateValidator();
```

Comment:
This whole thing feels way better. I doubt a user would need to implement custom logic on-chain for affiliates.